lib/faerie/src/backend.rs

```diff
@@ -59,8 +66,49 @@ impl FaerieBuilder {
             format,
             faerie_target,
             collect_traps,
+            libcall_names,
         })
     }
+
+    /// A default set of names for `ir::LibCall`. Indexed by `ir::LibCall` cast to an integer
+    /// representation. A function by this name is imported into the object as part of the
+    /// translation of a `ir::ExternalName::LibCall` variant. Calls to a LibCall should only be
+    /// inserted into the IR by the `cretonne_codegen` legalizer pass.
+    pub fn default_libcall_names() -> Vec<String> {
```

Comment:
I know using an array was my suggestion, but looking at it here, I'm worried that it's somewhat fragile, if we ever reorder the libcalls. Would you mind going with your idea, to pass in a closure, instead?

Reply:
I made this change. Its the first time I dealt with closure lifetimes like this so I'm not 100% sure I got it right, please let me know.